File: ios/EnterpriseMobileApp/Sources/Core/Domain/Entities/OfflineAction.swift
import Foundation

/// Queued action to sync when connectivity returns.
public struct OfflineAction: Equatable, Sendable, Identifiable {
    public let id: String
    public let actionType: OfflineActionType
    public let payload: [String: String]
    public let createdAt: Date
    public var synced: Bool

    public init(
        id: String = UUID().uuidString,
        actionType: OfflineActionType,
        payload: [String: String],
        createdAt: Date = Date(),
        synced: Bool = false
    ) {
        self.id = id
        self.actionType = actionType
        self.payload = payload
        self.createdAt = createdAt
        self.synced = synced
    }
}

public enum OfflineActionType: String, Equatable, Sendable {
    case addToCart
    case removeFromCart
    case redeemReward
    case sendMessage
}

File: android/core/domain/src/main/java/com/enterprise/core/domain/entities/OfflineAction.kt
package com.enterprise.core.domain.entities

import java.util.Date
import java.util.UUID

/** Queued action to sync when connectivity returns. */
data class OfflineAction(
    val id: String = UUID.randomUUID().toString(),
    val actionType: OfflineActionType,
    val payload: Map<String, String>,
    val createdAt: Date = Date(),
    val synced: Boolean = false
)

enum class OfflineActionType {
    ADD_TO_CART, REMOVE_FROM_CART, REDEEM_REWARD, SEND_MESSAGE
}
